integration_tests/tests/transactionCSV-to-s3-event-tests.ts

```diff
@@ -1,97 +1,70 @@
-import { waitForTrue } from "../../src/handlers/int-test-support/helpers/commonHelpers";
+import { ListObjectsCommandOutput } from "@aws-sdk/client-s3";
+import { poll } from "../../src/handlers/int-test-support/helpers/commonHelpers";
 import { resourcePrefix } from "../../src/handlers/int-test-support/helpers/envHelper";
-import { addFauxDataToTestCases } from "../../src/handlers/int-test-support/helpers/mock-data/csv/addFauxDataToTestCases";
-import { objectsToCSV } from "../../src/handlers/int-test-support/helpers/mock-data/csv/objectsToCsv";
-import { testCases } from "../../src/handlers/int-test-support/helpers/mock-data/csv/transformCSV-to-event-test-data";
+import { mockIdpCsvData } from "../../src/handlers/int-test-support/helpers/mock-data/csv";
 import {
   deleteS3Objects,
   getS3Object,
   listS3Objects,
   putS3Object,
-  S3Object,
 } from "../../src/handlers/int-test-support/helpers/s3Helper";
 
-const prefix = resourcePrefix();
-const folderPrefix = "btm_transactions";
-const bucketName = `${prefix}-storage`;
+const transactionsDirectory = "btm_transactions";
+const outputBucket = `${resourcePrefix()}-storage`;
+const inputBucket = `${resourcePrefix()}-transaction-csv`;
 
-const checkS3BucketForEventIds = async (): Promise<boolean> => {
-  const result = await listS3Objects({
-    bucketName,
-    prefix: folderPrefix,
-  });
-  if (
-    result.Contents === undefined ||
-    result.Contents.length !==
-      testCases.filter((data) => data.expectedPath === "happy").length
-  ) {
-    return false;
-  } else {
-    return true;
-  }
-};
```

Comment:
This is the function that used to be passed into waitForTrue. I've taken pretty much the same logic and passed it into this new `poll` function I've made which (I think) has a slightly more straight forward interface